crypto/s2n_kyber_evp.c

```diff
@@ -92,19 +92,19 @@ int s2n_kyber_evp_decapsulate(IN const struct s2n_kem *kem, OUT uint8_t *shared_
 int s2n_kyber_evp_generate_keypair(IN const struct s2n_kem *kem, OUT uint8_t *public_key,
         OUT uint8_t *secret_key)
 {
-    POSIX_BAIL(S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API);
+    POSIX_BAIL(S2N_ERR_UNIMPLEMENTED);
```

Comment:
I remembered a discussion on this: https://github.com/aws/s2n-tls/pull/4283#discussion_r1399727666

I'd say the same logic that led to S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API also means that S2N_ERR_UNIMPLEMENTED is fine.

Reply:
Agreed - I also think it's consistent with other feature gated code where the error should never be raised, since the code should never actually be called:
https://github.com/aws/s2n-tls/blob/ada65a76ac7154644f85024ed5ae4696f4409226/tls/s2n_prf.c#L614-L618